src/ray/object_manager/object_manager.cc

```diff
@@ -157,13 +161,9 @@ ray::Status ObjectManager::PullEstablishConnection(const ObjectID &object_id,
   // TODO(hme): There is no cap on the number of pull request connections.
   status = connection_pool_.GetSender(ConnectionPool::ConnectionType::MESSAGE, client_id,
                                       &conn);
-  if (!status.ok()) {
-    // TODO(hme): Keep track of retries,
-    // and only retry on object not local
-    // for now.
-    SchedulePull(object_id, config_.pull_timeout_ms);
-    return status;
-  }
+  // Currently, acquiring a connection should not fail.
+  RAY_CHECK_OK(status);
```

Comment:
Why should this not fail?

Reply:
Will add a note about why this should not fail.